File: utilities/python_utilities/main/src/pyspark/ConvertTimestamp.py
from pyspark.sql import SparkSession
spark = SparkSession.builder.master("local").appName("spark-python-utilites").getOrCreate()

from pyspark.sql.types import *
from pyspark.sql.column import *
from pyspark.sql.functions import *
from email.utils import parsedate_to_datetime

datestr = "Thu Sep 12 2019 15:58:30 GMT-0500 (hora estándar de Colombia)"
print(parsedate_to_datetime(datestr))

df = spark.createDataFrame(["Thu Sep 12 2019 15:58:30 GMT-0500 (hora estándar de Colombia)"], "string",).toDF("Date")

parsedate_to_datetime_udf = udf(parsedate_to_datetime, TimestampType())
df2 = df.withColumn("timestamp", parsedate_to_datetime_udf(col("Date")))

df2.show(20, False)

File: utilities/scala_utilities/src/main/scala/spark/ConvertTimestamp.scala
package spark

import java.text.SimpleDateFormat

import spark.testsession.SparkWrapper

object ConvertTimestamp extends App with SparkWrapper{
  import sparkSession.implicits._
  import org.apache.spark.sql.functions._

  val df = Seq(
    //("Thu Sep 12 2019 15:58:30 GMT-0500 (hora estándar de Colombia)")
      ("Thu Sep 12 2019 15:58:30")
  ).toDF("Date")

  val df2 = df.withColumn("timestamp_raw", regexp_replace(col("Date"), "(hora estándar de Colombia)", "" ))
  val df3 = df2.withColumn("timestamp",to_timestamp(col("timestamp_raw"), "dd/MM/yy HH:mm"))
  //val df3 = df2.withColumn("timestamp", col("timestamp_raw").cast("timestamp"))

  df3.show(20, false)
}
